src/main/scala/org/broadinstitute/hail/driver/ExportGenotypes.scala

```diff
@@ -46,22 +49,36 @@ object ExportGenotypes extends Command {
     val sas = vds.saSignature
 
     val symTab = Map(
-      "v" ->(0, TVariant),
-      "va" ->(1, vas),
-      "s" ->(2, TSample),
-      "sa" ->(3, sas),
-      "g" ->(4, TGenotype))
+      "v" -> (0, TVariant),
+      "va" -> (1, vas),
+      "s" -> (2, TSample),
+      "sa" -> (3, sas),
+      "g" -> (4, TGenotype))
 
     val ec = EvalContext(symTab)
 
-    val (header, fs) = if (cond.endsWith(".columns")) {
+    val (header, parseResults) = if (cond.endsWith(".columns")) {
       val (h, functions) = ExportTSV.parseColumnsFile(ec, cond, sc.hadoopConfiguration)
       (Some(h), functions)
     }
     else
       Parser.parseExportArgs(cond, ec)
 
-    val a = ec.a
+    Option(options.typesFile).foreach { file =>
+      writeTextFile(file, sc.hadoopConfiguration) { out =>
+        val sb = new StringBuilder
+        header
+          .getOrElse(parseResults.indices.map(i => s"_$i").toArray)
+          .zip(parseResults)
+          .iterator
+          .foreachBetween { case (name, (t, f)) =>
```

Comment:
Can you add an implicit conversion from Array[T] to RichIterable[T]?  Then you can axe the `.iterator`.  (You do this in multiple spots.)